knimin/lib/data_access.py

```diff
@@ -3083,28 +3114,31 @@ def edit_sample_plate(self, id, name, plate_type_id, email=None,
             TRN.add(sql, sql_args)
             TRN.execute()
 
-    def read_sample_plate(self, id):
+    def read_sample_plate(self, plate_id):
         """Reads properties of a sample plate
 
         Parameters
         ----------
-        id : int
+        plate_id : int
             ID of the sample plate to read
 
         Returns
         -------
         dict
             {name : str, plate_type_id : int, email : str,
-             created_on: datetime, notes : str}
+             created_on: datetime, notes : str, studies: list of int}
             Properties of the sample plate: name, plate type ID, who created
-            it, when it was created, and notes
+            it, when it was created, notes, and the studies plated
         """
         with TRN:
-            self._sample_plate_exists(id)
-            sql = """SELECT name, plate_type_id, email, created_on, notes
+            self._sample_plate_exists(plate_id)
+            sql = """SELECT name, plate_type_id, email, created_on, notes,
+                        array_agg(study_id ORDER BY study_id) as studies
                      FROM pm.sample_plate
-                     WHERE sample_plate_id = %s"""
-            TRN.add(sql, [id])
+                        JOIN pm.sample_plate_study USING (sample_plate_id)
```

Comment:
Yes, this table maps the sample plate to the studies being plated. This is needed so we can filter the list of samples to populate the plate.